src/main.rs

```diff
@@ -60,58 +70,63 @@ impl BFInterp {
                 Instruction::Load(load) => self.load(&load.address, &load.dest),
                 Instruction::Call(call) => self.call_func(call),
                 Instruction::Add(add) => {
-                    self.int_bin_operations(&add.operand0, &add.operand1, &add.dest, 0)
+                    self.int_bin_operations(&add.operand0, &add.operand1, &add.dest, BinOps::Add)
                 }
                 Instruction::FAdd(fadd) => {
-                    self.fl_bin_operations(&fadd.operand0, &fadd.operand1, &fadd.dest, 0)
+                    self.fl_bin_operations(&fadd.operand0, &fadd.operand1, &fadd.dest, BinOps::Add)
                 }
                 Instruction::FPExt(fpext) => {
                     self.fpext(&fpext.operand, &fpext.to_type, &fpext.dest)
                 }
                 Instruction::Sub(sub) => {
-                    self.int_bin_operations(&sub.operand0, &sub.operand1, &sub.dest, 1)
+                    self.int_bin_operations(&sub.operand0, &sub.operand1, &sub.dest, BinOps::Sub)
                 }
                 Instruction::Mul(mul) => {
-                    self.int_bin_operations(&mul.operand0, &mul.operand1, &mul.dest, 2)
+                    self.int_bin_operations(&mul.operand0, &mul.operand1, &mul.dest, BinOps::Mul)
                 }
                 Instruction::UDiv(udiv) => {
-                    self.int_bin_operations(&udiv.operand0, &udiv.operand1, &udiv.dest, 3)
+                    self.int_bin_operations(&udiv.operand0, &udiv.operand1, &udiv.dest, BinOps::Div)
```

Comment:
The `UDiv` and `SDiv` instructions result in the exact same call, so we can probably combine them:
```
Instruction::UDiv(div) | Instruction::SDiv(div) => { ...
```

Reply:
Doesn't seem to work as the patterns don't match. (First div is UDiv and second div is SDiv.)

Reply:
Oh, I see. Nevermind then. :-)